server/SupServer.java

```diff
@@ -0,0 +1,86 @@
+import java.io.*;
+import java.net.*;
+import java.util.*;
+
+public class SupServer {
+  
+  // userList is a hashmap which is used to store the user name and its printwriter
+  HashMap<String, PrintWriter> userList = new HashMap<String, PrintWriter>();
+  int clientIndex = 1;
+  
+  public class SupClientHandler implements Runnable {
+    BufferedReader reader;
+    Socket sock;
+    
+    public SupClientHandler(Socket clientSocket) {
+      try {
+        sock = clientSocket;
+        InputStreamReader isReader = new InputStreamReader(sock.getInputStream());
+        reader = new BufferedReader(isReader);
+      }
+      catch(Exception ex) {
+        ex.printStackTrace();
+      }
+    }
+    
+    public void run() {
+      String message;
+      try{
+        message = reader.readLine();
+        while(message != null) {
+          System.out.println("read" + message + " from " + Thread.currentThread().getName());
+          tellSomeone(message, Thread.currentThread().getName());
+        }
+      }
+      catch(Exception ex) {
+        ex.printStackTrace();
+      }
+    }
+  }
+  
+  public void start () {
+    try{
+      ServerSocket supServer = new ServerSocket(3000);
+      
+      // everytime receive a socket from a server, create a new thread. put the printwriter into userlist and set the name for client
+      while(true) {
+        Socket supClient = supServer.accept();
+        PrintWriter writer = new PrintWriter(supClient.getOutputStream());
```

Comment:
This has to be a 2 way connection, we need both in and out, but this only saved the out. We could create another Client object that keeps track of this socket?